// @ark-ui/solid/dist/chunk/34OZO53S.js
import { collapsible_exports, useCollapsibleContext } from './3HMZSTYX.js';
import { useRenderStrategyContext, splitRenderStrategyProps, RenderStrategyProvider } from './L7OSYHVR.js';
import { createSplitProps } from './ZMHI4GDJ.js';
import { ark } from './EPLBB4QN.js';
import { useEnvironmentContext } from './QODB3C3M.js';
import { useLocaleContext } from './HPDOBQ3V.js';
import { createContext } from './I5UG2QUO.js';
import { __export } from './ESLJRKWD.js';
import { createComponent, mergeProps as mergeProps$1 } from 'solid-js/web';
import { mergeProps, useMachine, normalizeProps } from '@zag-js/solid';
import { createMemo, splitProps, createUniqueId } from 'solid-js';
import * as accordion from '@zag-js/accordion';
export { anatomy } from '@zag-js/accordion';

// src/components/accordion/use-accordion-context.ts
var [AccordionProvider, useAccordionContext] = createContext({
  hookName: "useAccordionContext",
  providerName: "<AccordionProvider />"
});

// src/components/accordion/accordion-context.tsx
var AccordionContext = (props) => props.children(useAccordionContext());

// src/components/accordion/use-accordion-item-context.ts
var [AccordionItemProvider, useAccordionItemContext] = createContext({
  hookName: "useAccordionItemContext",
  providerName: "<AccordionItemProvider />"
});

// src/components/accordion/use-accordion-item-props-context.ts
var [AccordionItemPropsProvider, useAccordionItemPropsContext] = createContext({
  hookName: "useAccordionItemPropsContext",
  providerName: "<AccordionItemPropsProvider />"
});

// src/components/accordion/accordion-item.tsx
var AccordionItem = (props) => {
  const [itemProps, localProps] = createSplitProps()(props, ["value", "disabled"]);
  const accordion2 = useAccordionContext();
  const renderStrategyProps = useRenderStrategyContext();
  const mergedProps = mergeProps(() => accordion2().getItemProps(itemProps), localProps);
  const itemState = createMemo(() => accordion2().getItemState(itemProps));
  const itemContentProps = accordion2().getItemContentProps(itemProps);
  return createComponent(AccordionItemPropsProvider, {
    value: itemProps,
    get children() {
      return createComponent(AccordionItemProvider, {
        value: itemState,
        get children() {
          return createComponent(collapsible_exports.Root, mergeProps$1({
            get open() {
              return itemState().expanded;
            },
            get ids() {
              return {
                content: itemContentProps.id
              };
            }
          }, renderStrategyProps, mergedProps));
        }
      });
    }
  });
};
var splitVisibilityProps = createSplitProps();
var AccordionItemContent = (props) => {
  const accordion2 = useAccordionContext();
  const itemProps = useAccordionItemPropsContext();
  const itemContentProps = createMemo(() => {
    const contentProps = accordion2().getItemContentProps(itemProps);
    const [, ownProps] = splitVisibilityProps(contentProps, ["hidden", "data-state"]);
    return ownProps;
  });
  const mergedProps = mergeProps(() => itemContentProps(), props);
  return createComponent(collapsible_exports.Content, mergedProps);
};

// src/components/accordion/accordion-item-context.tsx
var AccordionItemContext = (props) => props.children(useAccordionItemContext());
var AccordionItemIndicator = (props) => {
  const accordion2 = useAccordionContext();
  const itemProps = useAccordionItemPropsContext();
  const mergedProps = mergeProps(() => accordion2().getItemIndicatorProps(itemProps), props);
  return createComponent(ark.div, mergedProps);
};
var AccordionItemTrigger = (props) => {
  const accordion2 = useAccordionContext();
  const itemProps = useAccordionItemPropsContext();
  const collapsible = useCollapsibleContext();
  const mergedProps = mergeProps(() => accordion2().getItemTriggerProps(itemProps), props);
  const [ariaControls, buttonProps] = splitProps(mergedProps, ["aria-controls"]);
  return createComponent(ark.button, mergeProps$1(buttonProps, () => !collapsible().unmounted && ariaControls));
};
var useAccordion = (props = {}) => {
  const id = createUniqueId();
  const locale = useLocaleContext();
  const environment = useEnvironmentContext();
  const context = createMemo(() => ({
    id,
    dir: locale().dir,
    value: props.defaultValue,
    getRootNode: environment().getRootNode,
    ...props
  }));
  const [state, send] = useMachine(accordion.machine(context()), {
    context
  });
  return createMemo(() => accordion.connect(state, send, normalizeProps));
};

// src/components/accordion/accordion-root.tsx
var AccordionRoot = (props) => {
  const [renderStrategyProps, accordionProps] = splitRenderStrategyProps(props);
  const [useAccordionProps, localProps] = createSplitProps()(accordionProps, ["collapsible", "defaultValue", "disabled", "id", "ids", "multiple", "onFocusChange", "onValueChange", "orientation", "value"]);
  const api = useAccordion(useAccordionProps);
  const mergedProps = mergeProps(() => api().getRootProps(), localProps);
  return createComponent(AccordionProvider, {
    value: api,
    get children() {
      return createComponent(RenderStrategyProvider, {
        value: renderStrategyProps,
        get children() {
          return createComponent(ark.div, mergedProps);
        }
      });
    }
  });
};
var AccordionRootProvider = (props) => {
  const [renderStrategyProps, accordionProps] = splitRenderStrategyProps(props);
  const [{
    value: accordion2
  }, localProps] = createSplitProps()(accordionProps, ["value"]);
  const mergedProps = mergeProps(() => accordion2().getRootProps(), localProps);
  return createComponent(AccordionProvider, {
    value: accordion2,
    get children() {
      return createComponent(RenderStrategyProvider, {
        value: renderStrategyProps,
        get children() {
          return createComponent(ark.div, mergedProps);
        }
      });
    }
  });
};

// src/components/accordion/accordion.ts
var accordion_exports = {};
__export(accordion_exports, {
  Context: () => AccordionContext,
  Item: () => AccordionItem,
  ItemContent: () => AccordionItemContent,
  ItemContext: () => AccordionItemContext,
  ItemIndicator: () => AccordionItemIndicator,
  ItemTrigger: () => AccordionItemTrigger,
  Root: () => AccordionRoot,
  RootProvider: () => AccordionRootProvider
});

export { AccordionContext, AccordionItem, AccordionItemContent, AccordionItemContext, AccordionItemIndicator, AccordionItemTrigger, AccordionRoot, AccordionRootProvider, accordion_exports, useAccordion, useAccordionContext, useAccordionItemContext };
