// @ark-ui/solid/dist/chunk/CTTG4GSG.js
import { ListCollection, TreeCollection, filePathToTree } from '@zag-js/collection';

// src/components/collection.ts

// ../../node_modules/@zag-js/store/dist/index.mjs
function glob() {
  if (typeof globalThis !== "undefined") return globalThis;
  if (typeof self !== "undefined") return self;
  if (typeof window !== "undefined") return window;
  if (typeof global !== "undefined") return global;
}
function globalRef(key, value) {
  const g = glob();
  if (!g) return value();
  g[key] || (g[key] = value());
  return g[key];
}
var refSet = globalRef("__zag__refSet", () => /* @__PURE__ */ new WeakSet());
globalRef("__zag__proxyStateMap", () => /* @__PURE__ */ new WeakMap());
function ref(obj) {
  refSet.add(obj);
  return obj;
}

// src/components/collection.ts
var createListCollection = (options) => ref(new ListCollection(options));
var createTreeCollection = (options) => ref(new TreeCollection(options));
var createFileTreeCollection = (paths) => ref(filePathToTree(paths));

export { createFileTreeCollection, createListCollection, createTreeCollection };
