// @ark-ui/solid/dist/chunk/LCESUDZB.js
import { splitRenderStrategyProps } from './L7OSYHVR.js';
import { createSplitProps } from './ZMHI4GDJ.js';
import { ark } from './EPLBB4QN.js';
import { createContext } from './I5UG2QUO.js';
import { createComponent, mergeProps as mergeProps$1 } from 'solid-js/web';
import { useMachine, normalizeProps, mergeProps } from '@zag-js/solid';
import { createSignal, createMemo, createEffect, Show } from 'solid-js';
import * as presence from '@zag-js/presence';

// src/components/presence/split-presence-props.ts
var splitPresenceProps = (props) => createSplitProps()(props, [
  "immediate",
  "lazyMount",
  "onExitComplete",
  "present",
  "unmountOnExit"
]);
var usePresence = (props) => {
  const [renderStrategyProps, context] = splitRenderStrategyProps(props);
  const [wasEverPresent, setWasEverPresent] = createSignal(false);
  const [state, send] = useMachine(presence.machine(context), {
    context
  });
  const api = createMemo(() => presence.connect(state, send, normalizeProps));
  createEffect(() => {
    const present = api().present;
    if (present) setWasEverPresent(true);
  });
  return createMemo(() => ({
    unmounted: !api().present && !wasEverPresent() && renderStrategyProps.lazyMount || renderStrategyProps.unmountOnExit && !api().present && wasEverPresent(),
    present: api().present,
    presenceProps: {
      ref: api().setNode,
      hidden: !api().present,
      "data-state": context.present ? "open" : "closed"
    }
  }));
};

// src/components/presence/presence.tsx
var Presence = (props) => {
  const [presenceProps, localProps] = splitPresenceProps(props);
  const api = usePresence(presenceProps);
  const mergedProps = mergeProps(() => api().presenceProps, localProps);
  return createComponent(Show, {
    get when() {
      return !api().unmounted;
    },
    get children() {
      return createComponent(ark.div, mergeProps$1(mergedProps, {
        "data-scope": "presence",
        "data-part": "root"
      }));
    }
  });
};

// src/components/presence/use-presence-context.ts
var [PresenceProvider, usePresenceContext] = createContext({
  hookName: "usePresenceContext",
  providerName: "<PresenceProvider />"
});

export { Presence, PresenceProvider, splitPresenceProps, usePresence, usePresenceContext };
