// @corvu/utils/dist/chunk/4XMQWCCT.js
import { noPointerEvents_default } from './R7BXXTGP.js';
import { outsidePointer_default } from './4AJDAG2F.js';
import { escapeKeyDown_default } from './WW7THO4B.js';
import { mergeProps } from 'solid-js';

var createDismissible = (props) => {
  const defaultedProps = mergeProps(
    {
      dismissOnEscapeKeyDown: true,
      dismissOnOutsidePointer: true,
      closeOnOutsidePointerStrategy: "pointerup",
      noOutsidePointerEvents: true
    },
    props
  );
  escapeKeyDown_default({
    enabled: defaultedProps.dismissOnEscapeKeyDown,
    onEscapeKeyDown: (event) => {
      defaultedProps.onEscapeKeyDown?.(event);
      if (!event.defaultPrevented) {
        defaultedProps.onDismiss("escapeKey");
      }
    }
  });
  outsidePointer_default({
    enabled: defaultedProps.dismissOnOutsidePointer,
    strategy: defaultedProps.closeOnOutsidePointerStrategy,
    ignore: defaultedProps.dismissOnOutsidePointerIgnore,
    onPointer: (event) => {
      defaultedProps.onOutsidePointer?.(event);
      if (!event.defaultPrevented) {
        const ctrlLeftClick = event.button === 0 && event.ctrlKey === true;
        const isRightClick = event.button === 2 || ctrlLeftClick;
        if (isRightClick) return;
        defaultedProps.onDismiss("pointerOutside");
      }
    },
    element: defaultedProps.element
  });
  noPointerEvents_default({
    enabled: defaultedProps.noOutsidePointerEvents
  });
};
var dismissible_default = createDismissible;

export { dismissible_default };
